Mathlib/RingTheory/PowerSeries/Basic.lean

```diff
@@ -1832,9 +1836,63 @@ theorem trunc_add (n) (φ ψ : PowerSeries R) : trunc n (φ + ψ) = trunc n φ +
     · rw [zero_add]
 #align power_series.trunc_add PowerSeries.trunc_add
 
+/--
+`R⟦X⟧` is notation for `PowerSeries R`,
+the semiring of formal power series in one variable over a semiring `R`.
+-/
+scoped notation:9000 R "⟦X⟧" => PowerSeries R
+
+theorem trunc_succ (f : R⟦X⟧) (n : ℕ) :
+    trunc n.succ f = trunc n f + Polynomial.monomial n (coeff R n f) := by
+  rw [trunc, Ico_zero_eq_range, sum_range_succ, trunc, Ico_zero_eq_range]
+
+theorem natDegree_trunc_lt (f : R⟦X⟧) (n) : (trunc (n + 1) f).natDegree < n + 1 := by
```

Comment:
Why is this lemma  in terms of n+1 rather than n?

Reply:
Thank you for the comment. I'm not sure if this is what you meant, but if I replace n+1 by n in both places, then I think the lemma would be false. trunc 0 f = 0, which has natDegree 0.

Would it be better to make n implicit?

Reply:
Oh right I'm being silly indeed (or just thinking about degree instead of natDegree)

Reply:
can you add the `degree` version as well actually. I think we should try to keep approximate feature parity between the two

Reply:
Done. Thanks for the suggestion.